sklearn/isotonic.py

```diff
@@ -375,6 +378,9 @@ def transform(self, T):
             The transformed data
         """
         T = as_float_array(T)
+        if hasattr(self, '_dtype'):
```

Comment:
Basically a way to check if there is a saved X. I could replace it with

```if hasattr(self, '_necessary_X_'):
    T = T.astype(self._necessary_X_._dtype, copy=False)
```

I could check if it makes a difference in terms of speed but it's probably minimal (just attribute accesses, right?)